Sources/Controllers/DashboardOnMap/MapSettings/MapSettingsGpxViewController.swift

```diff
@@ -0,0 +1,592 @@
+            }
+            
+            if let cell {
+                let distance = item.obj(forKey: "distance") as? String ?? ""
+                let time = item.obj(forKey: "time") as? String ?? ""
+                let waypointCount = item.obj(forKey: "waypointCount") as? String ?? ""
+                cell.titleLabel.text = item.title
+                cell.descriptionLabel.text = [distance, time, waypointCount].filter { !$0.isEmpty }.joined(separator: " · ")
+                let iconImage: UIImage
+                if let iconName = item.iconName, let image = UIImage(named: iconName) {
+                    iconImage = image
+                } else {
+                    iconImage = UIImage(named: "ic_custom_trip") ?? UIImage()
+                }
+                
+                cell.leftIconView.image = iconImage.withRenderingMode(.alwaysTemplate)
+                cell.leftIconView.tintColor = item.iconTintColor
+            }
+            return cell
+        } else if item.cellType == OALargeImageTitleDescrTableViewCell.getIdentifier() {
+            var cell = tableView.dequeueReusableCell(withIdentifier: OALargeImageTitleDescrTableViewCell.getIdentifier()) as? OALargeImageTitleDescrTableViewCell
+            if cell == nil {
+                let nib = Bundle.main.loadNibNamed(OALargeImageTitleDescrTableViewCell.getIdentifier(), owner: self, options: nil)
+                cell = nib?.first as? OALargeImageTitleDescrTableViewCell
+                cell?.selectionStyle = .none
+            }
+            
+            if let cell = cell {
+                cell.titleLabel?.text = item.title
+                cell.titleLabel?.accessibilityLabel = item.title
+                cell.descriptionLabel?.text = item.descr
+                cell.descriptionLabel?.accessibilityLabel = item.descr
+                cell.cellImageView?.image = UIImage.templateImageNamed(item.iconName)
+                cell.cellImageView?.contentMode = .scaleAspectFill
+                cell.cellImageView?.clipsToBounds = true
+                cell.cellImageView?.tintColor = item.iconTintColor
+                cell.button?.setTitle(item.obj(forKey: "buttonTitle") as? String, for: .normal)
+                cell.button?.accessibilityLabel = item.obj(forKey: "buttonTitle") as? String
+                cell.button?.removeTarget(nil, action: nil, for: .allEvents)
+                cell.button?.tag = indexPath.section << 10 | indexPath.row
+                cell.button?.addTarget(self, action: #selector(onCellButtonClicked(sender:)), for: .touchUpInside)
+            }
+            
+            outCell = cell
+            let update: Bool = outCell?.needsUpdateConstraints() ?? false
+            if update {
+                outCell?.setNeedsUpdateConstraints()
+            }
+        }
+        return outCell
+    }
+    
+    override func onRowSelected(_ indexPath: IndexPath?) {
+        guard let indexPath = indexPath, let gpx = getGpxForSelectedRow(at: indexPath) else { return }
```

Comment:
guard let indexPath,

Reply:
code must be correct: guard let indexPath = indexPath, let gpx = getGpxForSelectedRow(at: indexPath) else { return }

Reply:
guard let indexPath, let gpx = getGpxForSelectedRow(at: indexPath) else { return }